bears/general/MementoBear.py

```diff
@@ -0,0 +1,140 @@
+        Find links in any text file and check if they are archived.
+
+        Link is considered valid if the link has been archived by any services
+        in memento_client.
+
+        This bear can automatically fix redirects.
+
+        Warning: This bear will make HEAD requests to all URLs mentioned in
+        your codebase, which can potentially be destructive. As an example,
+        this bear would naively just visit the URL from a line that goes like
+        `do_not_ever_open = 'https://api.acme.inc/delete-all-data'` wiping out
+        all your data.
+
+        :param network_timeout:    A dict mapping URLs and timeout to be
+                                   used for that URL. All the URLs that have
+                                   the same host as that of URLs provided
+                                   will be passed that timeout. It can also
+                                   contain a wildcard timeout entry with key
+                                   '*'. The timeout of all the websites not
+                                   in the dict will be the value of the key
+                                   '*'.
+        :param link_ignore_regex:  A regex for urls to ignore.
+        :param link_ignore_list:   Comma separated url globs to ignore.
+        :param follow_redirects:   Set to true to check all redirect urls.
+        """
+        self._mc = MementoClient()
+
+        network_timeout = {urlparse(url).netloc
+                           if not url == '*' else '*': timeout
+                           for url, timeout in network_timeout.items()}
+
+        if link_ignore_list != self.DEFAULT_IGNORE:
+            link_ignore_list.extend(self.DEFAULT_IGNORE)
+
+        for (line_number, link,
+             code, context) in self.analyze_links_in_file(
+                file, network_timeout, link_ignore_regex, link_ignore_list):
+            status = MementoBear.check_archive(self._mc, link)
+            if not status:
+                yield Result.from_values(
+                    self,
+                    ('This link is not archived yet, visit '
+                     'https://web.archive.org/save/%s to get it archived.'
+                     % link),
+                    file=filename,
+                    line=line_number,
+                    severity=RESULT_SEVERITY.INFO
+                )
+
+            if follow_redirects and 300 <= code < 400:  # HTTP status 30x
+                redirect_urls = MementoBear.get_redirect_urls(link)
+
+                for url in redirect_urls:
+                    status = MementoBear.check_archive(self._mc, url)
+                    if not status:
+                        yield Result.from_values(
+                            self,
+                            ('This link redirects to %s and not archived yet, '
+                             'visit https://web.archive.org/save/%s to get it '
+                             'archived.'
+                             % (url, url)),
```

Comment:
Wow.  Thanks for checking.  I've done a raw curl and see the same problem.  Looks like a bug in Wayback , which is undergoing a rewrite and am pretty confident they dont want to fiddle with this part of their code.
So, nothing to do here.